backtrace-library/src/main/java/backtraceio/library/models/BacktraceExceptionHandler.java

```diff
@@ -80,4 +84,26 @@ private OnServerResponseEventListener getCallbackToDefaultHandler(final Thread t
             signal.countDown();
         };
     }
+
+    private List<BacktraceReport> transformExceptionIntoReports(Throwable throwable) {
+        final String exceptionTrace = UUID.randomUUID().toString();
+        BacktraceReport parent = null;
+
+        List<BacktraceReport> reports = new ArrayList<>();
+        while (throwable != null) {
+            Exception currentException = throwable instanceof Exception ? (Exception) throwable : new UnhandledThrowableWrapper(throwable);
+            BacktraceReport report = new BacktraceReport(currentException, BacktraceExceptionHandler.customAttributes);
+
+            report.attributes.put(BacktraceAttributeConsts.ErrorType, BacktraceAttributeConsts.UnhandledExceptionAttributeType);
+            report.attributes.put("error.trace", exceptionTrace);
+            report.attributes.put("error.id", report.uuid.toString());
```

Comment:
why do we need a duplicate uuid?

Reply:
uuid is not available in the attributes, to make it clear what the parent refers to, I think error.id makes sense here